### ISSUE TITLE
Builder backends are only exposed in the Internal module

### ISSUE THREAD

**Issue by nikita-volkov** [fumieval/mason#4]
Yet they are used in public signatures. E.g.,

```haskell
toStrictByteString :: BuilderFor GrowingBuffer -> ByteString
```

This means that to provide a signature for a function using this one the user will have to reach out for an internal module.

So could you either reexport the backend types in the public module or if you don't want to, could you at least expose aliases like the following?

```haskell
type GrowingBufferBuilder = BuilderFor GrowingBuffer
```

Seeing that all this confusion is caused by the use of Internal module, I want to express other arguments and try to [convince you to not expose internal modules](http://nikita-volkov.github.io/internal-convention-is-a-mistake/).

**Comment by fumieval** [comment]
I noticed the very problem and I added some type synonyms for backends a few days ago: https://github.com/fumieval/mason/commit/51523b73c879c3ca5181b5464817b3d5fd40bcc1

I've just released [mason-0.2.3](http://hackage.haskell.org/package/mason-0.2.3) including this change.

**Comment by nikita-volkov** [comment]
Thanks.
